plugins/destination/postgresql/pgarrow/to_arrow.go

```diff
@@ -240,3 +234,37 @@ func parseTime(t string) (arrow.DataType, bool) {
 		return arrow.FixedWidthTypes.Time64us, true
 	}
 }
+
+func parseNumeric(t string) (arrow.DataType, bool) {
+	matches := reNumeric.FindAllStringSubmatch(t, -1)
+	if len(matches) == 0 {
+		return nil, false
+	}
+
+	if len(matches[0]) < 3 || matches[0][1] == "" {
+		// no precision/scale specified
+		d, _ := arrow.NewDecimalType(arrow.DECIMAL256, 38, 0)
+		return d, true
+	}
+
+	precision, err := strconv.ParseInt(matches[0][1], 10, 64)
+	if precision == 0 || err != nil {
+		panic("precision cannot be 0")
+	}
+	scale, err := strconv.ParseInt(matches[0][2], 10, 64)
+	if err != nil {
+		panic("error parsing scale " + err.Error())
+	}
+
+	if precision == 20 && scale == 0 {
+		// special case
+		return arrow.PrimitiveTypes.Uint64, true
+	}
+
+	if precision <= 76 {
```

Comment:
If the prec is<=38 you can use decimal128, btw

Reply:
Removed in https://github.com/cloudquery/cloudquery/pull/17247/commits/65118a48342d563dbc9ca365b5a347b0876a4cb9 to simplify